configs/config-tfm.h

```diff
@@ -0,0 +1,47 @@
+/**
+ * \file config-tfm.h
+ *
+ * \brief TF-M configuration with tweaks for a successful build and test.
+ */
+/*
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/* TF-M Configuration Options */
+#include "../configs/ext/tfm_mbedcrypto_config_profile_medium.h"
+
+/* TF-M PSA Crypto Configuration */
+#define MBEDTLS_PSA_CRYPTO_CONFIG_FILE "../configs/ext/crypto_config_profile_medium.h"
+
+/*****************************************************************************/
+/* Tweak configuration based on TF-M config for a successful build and test. */
+/*****************************************************************************/
+
+/* MBEDTLS_PSA_CRYPTO_SPM needs third party files, so disable it. */
+#undef MBEDTLS_PSA_CRYPTO_SPM
+/* TF-M provides its own (dummy) implemenations which Mbed TLS doesn't need. */
+#undef MBEDTLS_AES_SETKEY_DEC_ALT
+#undef MBEDTLS_AES_DECRYPT_ALT
+/* pkparse.c fails to link without this. */
+#define MBEDTLS_OID_C
```

Comment:
How is this working for TF-M?

Also, it looks like we're missing a check in `check_config.h`.

Reply:
It's not working for TF-M unless the linker eliminates the pk parse/write functions, i.e. the functions are compiled but can't be linked. Since we copied the file, TF-M have [removed PK_PARSE and PK_WRITE from the profile](https://git.trustedfirmware.org/TF-M/trusted-firmware-m.git/commit/lib/ext/mbedcrypto/mbedcrypto_config/tfm_mbedcrypto_config_profile_medium.h?id=8a5469cb2469cbddc2977003aa4b6e480503a374).

Reply:
We shouldn't add the OID module to code size measurements: it's not something TF-M wants. Let's wait to add this line until we've updated the TF-M config file.

Reply:
Filed for later: https://github.com/Mbed-TLS/mbedtls/issues/8167

Reply:
@yanrayw FYI I've updated our copy of their config: https://github.com/Mbed-TLS/mbedtls/pull/8243